### ISSUE TITLE
Insurance for Player-Owned Ships

### ISSUE THREAD

**Issue by Ferociousfeind** [endless-sky/endless-sky#8128]
### Problem Description

Losing a ship you just bought can be not only disheartening, but also a total hassle to rebuild from scratch. Especially if you fly around the galaxy and custom-build your ultimate ships, losing them can represent a huge wasted time investment.

This issue, of losing ships you've poured time into perfecting, is a very large part of what makes fighter carriers unviable and completely unusable- they're good for one fight, and one fight only!

### Related Issue Links

#3920 Suggestion: What if fighters & drones are defeated instead of destroyed?
#6861 Feature request: Player-owned fighters could be bought back as part of a salvaging mechanic

### Desired Solution

An insurance system, which I have outlined once or twice before-

In the "hire" or maybe the "bank" tab, you can purchase certain types of insurance for your various ships. Under the hood, the "insurance" object (in the same way a system is an "object") would specify certain criteria, like the sign-on fee, the daily insurance fee, and the buyback fee. Maybe also the credit score you need to qualify. And also (optionally, like all the other criteria) shipyards and outfitters which restrict which individual ships can be insured.

(If there are no outfitters specified, all outfits will be insured. If there are no shipyards specified, all ship model names (which are saved to the save file) will be insured. If no costs are specified, the insured ship(s) can be recovered for free. Et cetera)

Say you bought a new Bulk Freighter for 12 million, 500 thousand credits. You are paranoid, and really want it insured. Undoubtedly, pirates will attempt to destroy it. So, you go to the bank tab, and purchase, hmm, "Ultra Premium Insurance", paying, say, 10% of the ship's cost up front (1.25M credits) and agreeing to a 0.05% daily insurance payment (which would be 6,250 credits), and if your ship is destroyed, you'll be charged a 1% recovery fee (of 125,000 credits).

You fly into dangerous space, paying the equivalent of 62.5 crew each day to get to the pirate-infested system, and disaster strikes. A swarm of Furies painted in red and black ambush you! You escape in your speedy Scout or whatever, but your poor Bulk Freighter didn't make it. You fly to the nearest system in tears. Your pride and joy! But you have that one trick up your sleeve.

You enter the shipyard, and are prompted with a dialog box, like the "pay XXX credits to restock all ships?" box. You can pay 125 thousand credits now to receive a new Bulk Freighter, with the same name and the same internals and everything. At the end of about a week, you've spent 13,918,750 credits on what could've been a 25,000,000 credit expense. Or, ignoring the initial 12.5M expense, you've spent 1.4M credits instead of 12.5M credits a second time.

And, of course, the credit specifics of any insurance system could be customized. Maybe it's free (and compulsory) to sign up for, with no daily cost, and a basic 5% cost to recover a lost ship. That'd mean spending an additional 625,000 credits instead of an additional 12.5M. Maybe the whole thing is free, and losing ships is only a temporary inconvenience? Maybe an insurance system is fine-tuned to incur a daily 2500 credit cost, to replace or emulate the starting loan.

Something to consider is, how does a ship's base price (not even counting depreciation) changing over time affect this? Especially the up-front cost, and also the rebuy cost. Do you save money by selling all the outfits on a ship and then buying insurance against its bare hull, and piling on the expensive outfits afterwards? Or do outfits purchased onto an insured ship with a sign-on fee also incur that same fee on themselves? I.e. if you bought 10% up-front insurance on your Arfecta, and then purchased a Jump Drive for it, would that cost 1M credits, or 1.1M credits? Would moving a JD onto it be free still, or would that cost 100K credits? Would moving the JD off of it reward you with 100K credits on the other hand? These are questions to consider, otherwise you may be able to commit Insurance Fraud...

### Alternative Approaches

There are multiple different approaches to fixing the issue of losing ships, especially fighters. A plethora have been suggested before. As for specifically alternative approaches to insurance... Maybe a simpler hardcoded "if a ship is ever lost, it can be repurchased in its moments-before-death form at a steep discount" insurance system? Though I would prefer the freedom of data file defined insurance systems, the way we have data file defined governments, and legalities, and loans.

### Additional Context

I am opening this issue instead of making it as a PR with my own ideas because I have _no_ idea how to approach... _any_ of this. Not a clue. I've got some data file syntax pretty much ready to go, and I've thought a little bit on the consequences of rebuying ships at discounts could be (not _very_ much, since you're not getting anything _new_, you're just preventing the loss of something you already had), but that's it.

**Comment by Zitchas** [comment]
I like the idea of having insurance... But I don't like the idea of it being able to miraculously replace outfits that, to be blunt, the bank/insurance agent don't have the faintest clue of even *where* to find it, let alone have permissions to do so. For instance, that Jump Drive you put on the Bulk Freighter... If the bank/insurance agency can just magically replace that, then there's a whole lot of officials during the Pug invasion that would like to have a word with them. Not to mention me, because I was forced to take on the pug with just seven ships. I'd have happily looted the bank to get an eighth...

But replacing a ship for whom the bank has access to a shipyard, and outfits for which they have access to the outfitter? Sure, definitely! I could see a company in human space being able to source anything available anywhere in human space, for instance. So at least that'd save the player having to go hither and yon across human space to re-equip their ship.

**Comment by Ferociousfeind** [comment]
Yeah, that's why I mention specifying outfitters/shipyards- if an outfit can't be found in XYZ collection of outfitters, then it can't be insured because "what even is this?", either in the form of denying insurance and preventing equipping of un-insurable outfits, or allowing insurance but letting the player know that XYZ outfits can't be insured and will be lost. (And also source governments, so that the hai aren't forced to recover your ship, and only the republic or the syndicate or the FW shipyards have to)

And, of course, un-insurable ships simply can't be insured, outfits on uninsured ships are all lost (otherwise where would they go? in a care package?)

Though, we have to acknowledge that not insuring korath or ka'het or wanderer outfits will keep fighters and drones and small ships from being late-game viable- you'll still lose the outfits. Even if it's immersion-breaking, is it _that_ bad to apologize to the player and fix their rare exotic toys?

**Comment by johndh** [comment]
Since the galaxy clearly has a robust manufacturing and shipping industry, what if you could just custom order a ship with whatever outfits are available in known space and pay to have it shipped to you and/or pick it up from one of the big centers like Zug or Polaris. I don't like having to go all the way across the galaxy to get heavy laser turrets, and would gladly pay an outrageous markup to have them shipped to one of my dominated pirate worlds. This also opens up more delivery and escort quests, delivering outfits and escorting ships (that aren't freighters) to wealthy captains.

This would still make "special" outfits and ships special, since nobody makes jump drives or a 512, etc.

**Comment by FlightSimEnjoyer** [comment]
I think that this could be implemented in the following way:
Some planets should have a tab on their menu for the insurance company. You can choose to pay a daily fee for insurance and in exchange the insurance company gives you the value of the insured spaceships if the ships are destroyed. 

If the ship is alien, maybe the insurance company would give the player less money for the insurance payment then the market value of the ship or maybe it would refuse to insure it. This would solve the problem outlined by Ferociousfeind of some ships being late-game unviable.

I'll try to implement this during the next few weeks.

**Comment by Ferociousfeind** [comment]
> I think that this could be implemented in the following way: Some planets should have a tab on their menu for the insurance company. You can choose to pay a daily fee for insurance and in exchange the insurance company gives you the value of the insured spaceships if the ships are destroyed.
> 
> If the ship is alien, maybe the insurance company would give the player less money for the insurance payment then the market value of the ship or maybe it would refuse to insure it. This would solve the problem outlined by Ferociousfeind of some ships being late-game unviable.
> 
> I'll try to implement this during the next few weeks.

I'd like to make it clear that the money isn't _exactly_ the issue with certain classes of ship being unviable... it's overwhelmingly the hassle in re-acquiring them once they've been destroyed. While in truth an insurance company would likely only be willing to reimburse the value of lost ships in the form of paltry credits, the player can get those whenever they want, basically however they want. But, for example, the Emerald Sword is a one-time ship that cannot be bought. If it dies, it's gone forever. Same with its fighters (by god its fighters...) and limited outfits like the cloaking device. Why ever use those fighters? Why ever put the cloak on something other than the flagship? You'll just lose it!

(So thr solution I'm looking for is specifically "you lost a ship, but now you can have it back because XYZ")

Nevertheless, I'll watch for your work in the coming weeks.

**Comment by johndh** [comment]
By the time you have unique or alien ships, you basically have unlimited money.
